src/pybind/mgr/volumes/volume.py

```diff
@@ -0,0 +1,151 @@
+                   'data': data_pool}
+        return self.mgr.mon_command(command)
+
+    def remove_filesystem(self, fs_name):
+        command = {'prefix': 'fs rm', 'fs_name': fs_name, 'yes_i_really_mean_it': True}
+        return self.mgr.mon_command(command)
+
+    def create_mds(self, fs_name):
+        spec = orchestrator.StatelessServiceSpec()
+        spec.name = fs_name
+        try:
+            completion = self.mgr.add_stateless_service("mds", spec)
+            self.mgr._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+        except (ImportError, orchestrator.OrchestratorError):
+            return 0, "", "Volume created successfully (no MDS daemons created)"
+        except Exception as e:
+            # Don't let detailed orchestrator exceptions (python backtraces)
+            # bubble out to the user
+            return -errno.EINVAL, "", str(e)
+        return 0, "", ""
+
+    def teardown_mds(self, fs_name):
+        command = {'prefix': 'fs set', 'fs_name': fs_name, 'var': 'cluster_down', 'val': 'true'}
+        r, outb, outs = self.mgr.mon_command(command)
+        if r != 0:
+            return r, outb, outs
+        for mds in self.get_mds_names(fs_name):
+            command = {'prefix': 'mds fail', 'role_or_gid': mds}
+            r, outb, outs = self.mgr.mon_command(command)
+            if r != 0:
+                return r, outb, outs
+        return 0, "", ""
+
+    ### volume operations -- create, rm, ls
+
+    def create_volume(self, volname, size=None):
+        """
+        """
+        metadata_pool, data_pool = self.gen_pool_names(volname)
+        # create pools
+        r, outs, outb = self.create_pool(metadata_pool, 16, pg_num_min=16, pg_autoscale_factor=4.0)
+        if r != 0:
+            return r, outb, outs
+        r, outb, outs = self.create_pool(data_pool, 8)
+        if r != 0:
+            return r, outb, outs
+        # create filesystem
+        r, outb, outs = self.create_filesystem(volname, metadata_pool, data_pool)
+        if r != 0:
+            return r, outb, outs
+        # create mds
+        return self.create_mds(volname)
+
+    def delete_volume(self, volname):
+        """
+        """
+        # Tear down MDS daemons
+        try:
+            completion = self.mgr.remove_stateless_service("mds", volname)
+            self.mgr._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+        except (ImportError, orchestrator.OrchestratorError):
+            pass
+        except Exception as e:
+            # Don't let detailed orchestrator exceptions (python backtraces)
+            # bubble out to the user
+            return -errno.EINVAL, "", str(e)
+
+        # In case orchestrator didn't tear down MDS daemons cleanly, or
+        # there was no orchestrator, we force the daemons down.
+        if self.volume_exists(volname):
+            r, outb, outs = self.teardown_mds(volname)
```

Comment:
The naming of the method is confusing. Here you don't teardown MDS, you set the mds daemons down. Please rename this method.